domains/client/domain-executor/src/lib.rs

```diff
@@ -221,6 +221,7 @@ where
         let active_leaves = active_leaves(primary_chain_client.as_ref(), select_chain).await?;
 
         let bundle_producer = BundleProducer::new(
+            DomainId::SYSTEM,
```

Comment:
I'm wondering if this should have also been an argument on `fn new` above, so that the whole domain executor is agnostic to domain ID.

Reply:
I'm thinking about copying the executor as SystemExecutor and deriving CoreExecutor from SystemExecutor as there are some nuances I'm unsure about between the system and core domain executors. Once we can make the initial core domain executor, we could think about a nice abstraction to refactor them.